tsdb/wal/wal.go

```diff
@@ -774,6 +784,52 @@ func listSegments(dir string) (refs []segmentRef, err error) {
 	return refs, nil
 }
 
+// Get segments from the directory
+func getSegments(dir string) ([]segmentRef, error) {
+	files, err := fileutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+	var refs []segmentRef
+	for _, fn := range files {
+		k, err := strconv.Atoi(fn)
+		if err != nil {
+			return nil, err
+		}
+		refs = append(refs, segmentRef{name: fn, index: k})
+	}
+	sort.Slice(refs, func(i, j int) bool {
+		return refs[i].index < refs[j].index
+	})
+	return refs, nil
+}
+
+// RepairUnsequentialSegments repairs the out of sequence segments by deleting the onces after the sequence gap.
+// Returns the sequential segments after deleting the out of sequence wals after the sequence gap.
```

Comment:
I got a couple of questions here for maintainers:
1. Should we just recover data from all segments without deleting any of them?
2. Why delete the segments after the gap and not before the gap?

/cc @krasi-georgiev @brian-brazil